extmod/modussl_axtls.c

```diff
@@ -277,6 +277,19 @@ STATIC mp_uint_t ussl_socket_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_t a
         ssl_free(self->ssl_sock);
         ssl_ctx_free(self->ssl_ctx);
         self->ssl_sock = NULL;
+    } else if (request == MP_STREAM_POLL) {
+        // If the caller is only interested in reading, and we have data pending, return immediately
+        if (arg == MP_STREAM_POLL_RD && self->bytes_left > 0) {
+            return MP_STREAM_POLL_RD;
+        }
+
+        // ...otherwise fall through to pass request to underlying socket
+        mp_uint_t ret = mp_get_stream(self->sock)->ioctl(self->sock, request, arg, errcode);
```

Comment:
To reduce code size, it would be better to have just one call to `->ioctl()`, ie combine this call with the one at the end of the function.